### PR TITLE
[Buttons] Deprecate customTitleColor and remove auto-accessibility from setTitleColor

### PR DESCRIPTION
**Pull request by samnm**
This deprecates customTitleColor on MDCButton and removes the automatic title color accessibility  magic from setTitleColor:forState:.

Currently setTitleColor:forState will highjack the title color if it isn't of sufficient contrast with the button background color. This PR undoes this behavior. If you were dependent on this feature please refer to MDCButtonTitleColorAccessibilityMutator.

### CODE REVIEW

**Review by ajsecord — APPROVED** [review]
Please merge or drop.

**Comment by samnm** [comment]
I'm waiting on feedback from the internal discussion before merging.

**Pull request merged**
